src/act/node/candid/type_arg.rs

```diff
@@ -0,0 +1,39 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+
+use crate::{
+    act::node::Context,
+    traits::{ToTokenStream, ToTypeAnnotation},
+};
+
+use super::CandidType;
+
+#[derive(Clone, Debug)]
+pub struct TypeArg(pub CandidType);
+
+impl TypeArg {
+    pub fn get_inline_name(&self, inline_name: &str, index: usize) -> String {
+        format!("{inline_name}TypeArg{index}")
+    }
+}
+
+impl ToTokenStream<Context> for Vec<TypeArg> {
+    fn to_token_stream(&self, context: &Context, inline_name: &str) -> proc_macro2::TokenStream {
+        let type_argument_token_streams: Vec<TokenStream> = self
+            .iter()
+            .enumerate()
+            .map(|(index, type_argument)| {
+                type_argument
+                    .0
```

Comment:
What if you did the dereferencing trick here. See ReturnType which does essentially the same thing